boilerplate/openshift/osd-container-image/README.md

```diff
@@ -0,0 +1,31 @@
+# Conventions for OSD Containers
+
+This convention is suitable for standalone containers - if an operator is desired other conventions should be used.
+
+> Note: The repository's main `Makefile` needs to be edited to have the following line:
+
+```make
+include boilerplate/generated-includes.mk
+```
+
+## `make` targets and functions
+
+The provided `Makefile` will interact build and push a container image defined by a Dockerfile at `build/Dockerfile`. If multiple containers are contained in the repo, they can also be managed by defining an `ADDITIONAL_IMAGE_SPECS` variable like so:
+
+```make
+define ADDITIONAL_IMAGE_SPECS
+./path/to/a/Dockerfile $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)/a-image:v1.2.3
+./path/to/b/Dockerfile $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)/b-image:v4.5.6
+endef
+```
+
+| Makefile target | Description |
+|---|---|
+| `make osd-container-image-build` | Builds the default container at `build/Dockerfile` and tag it based on the commit. Specify `DOCKERFILE` and `IMAGE_URI` to build other containers. |
+| `make osd-container-image-push` | Push the default container. |
+| `make osd-container-image-build-push` | Builds and pushes the default container and `ADDITIONAL_IMAGE_SPECS`, meant to be run by app-interface. |
+| `make isclean` | Ensures the local git checkout is clean. |
+
+## Linting/Testing
+
+This boilerplate convention does not contain any linting or testing guidelines to support a variety of containers. Those `Makefile` targets should be defined by the consumer themselves.
```

Comment:
I can't find a way to reply to your top-level comment, so I'm using this as the most relevant review comment 😅 

> Are you planning to create another convention for "go binary" type stuff (go-build, go-check, go-test, go-generate?

I was not planning on it... mostly because when I was looking at the existing repos there didn't seem to be agreement. * osd-network-verifier [inlines go build in the Dockerfile](https://github.com/openshift/osd-network-verifier/blob/b71e3eac178340e476a2d3a6894813115d15b3e0/build/Dockerfile)
* osdctl uses [goreleaser](https://github.com/openshift/osdctl/blob/master/.goreleaser.yml#L7-L31)
* managed-cluster-validating-webhooks and ocm-agent are probably the most "golang-osd-operator-esque" [mcvm](https://github.com/openshift/managed-cluster-validating-webhooks/blob/master/Makefile#L25-L31) & [ocm-agent](https://github.com/openshift/ocm-agent/blob/master/Makefile#L23-L28), but even they have slight differences just in how the repos are structured (cmd/main.go vs cmd/ocm-agent/main.go)

> E.g. it would be nice for osd-cluster-ready to be able to import that convention along with this one so its dockerfile can invoke make go-build rather than inlining the build command.

But nonetheless I agree with you that it would be so nice! After I saw the different approaches, I felt it was important to get a foundation that all of them could use and then potentially iterate on this convention (or a separate convention)

Reply:
Gotcha.

IMO including nice generic go-build and related targets in this or a separate convention would be a good look even if some of the consuming repos decide to ignore them. Yes, it can be done later; but the sooner it's in place, the higher the probability of consuming repos taking advantage of it.

(I'm envisioning a future where golang-osd-operator itself either doesn't exist or exists only as a shell that transitively imports the necessary component conventions like this one. In that world, there are at least ~15 consuming repos all using the exact same targets.)

Reply:
That's a good point about how consuming repos can just ignore them - I'll add them in!

+1 to raising the probability consuming repos use it too